### PR TITLE
test: i2s: skip some cases that do not support in NXP platform

### PR DESCRIPTION
**Pull request by hakehuang**
user mode and state test are not support well in NXP platform

skip for now

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

### CODE REVIEW

**Review by galak — COMMENTED** [review]
Can you explain further why these need to be skipped for NXP.  Seems like the ifdef should be something not driver specific.

File: tests/drivers/i2s/i2s_api/src/main.c [hunk 1]
```diff
@@ -45,11 +45,14 @@ void test_main(void)
 			ztest_unit_test(test_i2s_rx_transfer_configure_1),
 			ztest_unit_test(test_i2s_state_not_ready_neg),
 			ztest_unit_test(test_i2s_state_ready_neg),
+#ifndef CONFIG_I2S_MCUX_FLEXCOMM
 			ztest_unit_test(test_i2s_state_running_neg),
 			ztest_unit_test(test_i2s_state_stopping_neg),
 			ztest_unit_test(test_i2s_state_error_neg));
+#endif
```
Comment (mmahadevan108) [inline tests/drivers/i2s/i2s_api/src/main.c:52]: We only need to skip test_i2s_state_running_neg

Comment (hakehuang) [reply tests/drivers/i2s/i2s_api/src/main.c:52]: ok

**Comment by hakehuang** [comment]
> Can you explain further why these need to be skipped for NXP. Seems like the ifdef should be something not driver specific.

@galak please check the PR comments https://github.com/zephyrproject-rtos/zephyr/pull/32054

File: tests/drivers/i2s/i2s_speed/src/test_i2s_speed.c [hunk 2]
```diff
@@ -223,6 +223,7 @@ void test_i2s_transfer_short(void)
  */
 void test_i2s_transfer_long(void)
 {
+#ifndef CONFIG_I2S_MCUX_FLEXCOMM
```
Comment (mmahadevan108) [inline tests/drivers/i2s/i2s_speed/src/test_i2s_speed.c:226]: This should not be required

Comment (hakehuang) [reply tests/drivers/i2s/i2s_speed/src/test_i2s_speed.c:226]: ok

**Review by mmahadevan108 — APPROVED** [review]

File: tests/drivers/i2s/i2s_api/src/main.c [hunk 3]
```diff
@@ -50,6 +50,7 @@ void test_main(void)
 			ztest_unit_test(test_i2s_state_error_neg));
 	ztest_run_test_suite(i2s_states_test);
 
+#ifndef CONFIG_I2S_MCUX_FLEXCOMM
```
Comment (anangl) [inline tests/drivers/i2s/i2s_api/src/main.c:53]: The issue with the tests in user mode (an MPU Fault) is not specific to this driver. #35590 should fix it.

Comment (hakehuang) [reply tests/drivers/i2s/i2s_api/src/main.c:53]: thanks @anangl

**Pull request closed without merging**